benchmarking/commons/benchmark_definitions/yahpo.py

```diff
@@ -0,0 +1,477 @@
+    "Amazon_employee_access": "34539",
+    "adult": "126025",
+    "nomao": "126026",
+    "bank-marketing": "126029",
+    "shuttle": "146212",
+    "Australian": "167104",
+    "kr-vs-kp": "167149",
+    "mfeat-factors": "167152",
+    "credit-g": "167161",
+    "vehicle": "167168",
+    "kc1": "167181",
+    "blood-transfusion-service-center": "167184",
+    "cnae-9": "167185",
+    "phoneme": "167190",
+    "higgs": "167200",
+    "connect-4": "167201",
+    "helena": "168329",
+    "jannis": "168330",
+    "volkert": "168331",
+    "MiniBooNE": "168335",
+    "APSFailure": "168868",
+    "christine": "168908",
+    "fabert": "168910",
+    "airlines": "189354",
+    "jasmine": "189862",
+    "sylvine": "189865",
+    "albert": "189866",
+    "dionis": "189873",
+    "car": "189905",
+    "segment": "189906",
+    "Fashion-MNIST": "189908",
+    "jungle_chess_2pcs_raw_endgame_complete": "189909",
+}
+
+
+def yahpo_nb301_benchmark(dataset_name):
+    return BenchmarkDefinition(
+        max_wallclock_time=1800,  # TODO
+        n_workers=4,
+        elapsed_time_attr="runtime",
+        metric="val_accuracy",
+        mode="max",
+        blackbox_name="yahpo-nb301",
+        dataset_name=dataset_name,
+        max_resource_attr="max_resources",
+    )
+
+
+yahpo_nb301_instances = ["CIFAR10"]
+
+
+yahpo_nb301_benchmark_definitions = {
+    "yahpo-nb301-" + name: yahpo_nb301_benchmark(name) for name in yahpo_nb301_instances
+}
+
+
+def yahpo_lcbench_benchmark(dataset_name):
+    return BenchmarkDefinition(
+        max_wallclock_time=7200,
+        n_workers=4,
+        elapsed_time_attr="time",
+        metric="val_accuracy",
+        mode="max",
+        blackbox_name="yahpo-lcbench",
+        dataset_name=openml_task_name_to_id[dataset_name],
+        max_resource_attr="max_resources",
+    )
+
+
+yahpo_lcbench_instances = list(openml_task_name_to_id.keys())
+
+
+yahpo_lcbench_benchmark_definitions = {
+    "yahpo-lcbench-" + name: yahpo_lcbench_benchmark(name)
+    for name in yahpo_lcbench_instances
+}
+
+
+yahpo_fcnet_instances = {
+    "naval": "fcnet_naval_propulsion",
+    "protein": "fcnet_protein_structure",
+    "slice": "fcnet_slice_localization",
+    "parkinsons": "fcnet_parkinsons_telemonitoring",
+}
+
+
+# TODO: "runtime" is NOT elapsed time, but total training time
+def yahpo_fcnet_benchmark(dataset_name):
+    return BenchmarkDefinition(
+        max_wallclock_time=1200,
+        n_workers=4,
+        elapsed_time_attr="runtime",
+        metric="valid_loss",
+        mode="min",
+        blackbox_name="yahpo-fcnet",
+        dataset_name=yahpo_fcnet_instances[dataset_name],
+        max_resource_attr="max_resources",
+    )
+
+
+yahpo_fcnet_benchmark_definitions = {
+    "yahpo-fcnet-" + name: yahpo_fcnet_benchmark(name) for name in yahpo_fcnet_instances
+}
+
+
+def yahpo_iaml_rpart_benchmark(dataset_name, metric, mode="max"):
+    return BenchmarkDefinition(
+        max_wallclock_time=1800,  # TODO
```

Comment:
Optional comment for this PR: I would recommend to set a bound on `max_num_evaluations`
https://github.com/awslabs/syne-tune/blob/main/syne_tune/stopping_criterion.py#L35

For instance, one can set `max_num_evaluations` to `100 x max_t` which ensures that 100 full configurations are seen for fifo schedulers and a similar number of evaluation is seen for hyperband schedulers.

Reply:
In the next PR, I am providing better numbers for these defaults. But I'll also add max_num_evaluations

Reply:
Note: yahpo.py will change quite a bit in the next PR for iaml and rbv2, because the way it is done here is cumbersome to work with.